### PR TITLE
Automatic Labeling for Dependencies, Security and Design

### LINKED ISSUE #2555: MOBILE: Automatic Labeling for Dependencies, Security and Design Related Issues
**Is your feature request related to a problem? Please describe.**

1. We spend a lot of time manually labeling issues that are created.
2. Automation will significantly reduce this overhead

**Describe the solution you'd like**

Using the `auto-label` GitHub action that we currently use: https://github.com/marketplace/actions/auto-label

1. Make configuration changes to the existing `issue.yml` file. 
1. Do not rename the existing `issue.yml` file.
1. **Dependencies**
    1. When the words `dependabot`, `dependency` or `dependencies` are in the body of the issue 
    2. Add the labels `dependencies`, `security` and `good first issue`
1. **Security**
    1. When the word `security` are in the body of the issue 
    2. Add the label `security` and `good first issue`
1. **Design**
    1. When the words `layout`, `screen`, `design` or `figma` are in the body of the issue 
    2. Add the label `ui/ux` and `good first issue`

**Describe alternatives you've considered**

- See above

**Approach to be followed (optional)**

- See above
- No other functionality must be affected

**Additional context**

Related Issues. The same person must be assigned to both:

- https://github.com/PalisadoesFoundation/talawa-admin/issues/2256
- https://github.com/PalisadoesFoundation/talawa/issues/2555

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by VanshikaSabharwal**
**What kind of change does this PR introduce?**

  security

**Issue Number:**

Fixes #2555 

**Did you add tests for your changes?**

 No

**Snapshots/Videos:**

talawa-issue-labelling-ss

**Summary**

**Does this PR introduce a breaking change?**

No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes 

## Summary by CodeRabbit

- **New Features**
	- Introduced automatic labeling for issues based on specific keywords, enhancing issue organization and prioritization.
	- Added a mapping for label synonyms to improve the flexibility of the auto-labeling feature.
	- Implemented a new workflow to automate the labeling of newly opened issues.
- **Style**
	- Updated formatting for consistency in the issues workflow configuration.

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please use the methodology listed here with the JSON file and the latest version of the GitHub action:
- https://github.com/marketplace/actions/auto-label

**Comment by VanshikaSabharwal** [comment]
> Please use the methodology listed here with the JSON file and the latest version of the GitHub action:
> 
> * https://github.com/marketplace/actions/auto-label

Okay I am doing it

**Comment by VanshikaSabharwal** [comment]
> Please use the methodology listed here with the JSON file and the latest version of the GitHub action:
> 
> * https://github.com/marketplace/actions/auto-label

I have made changes is it fine now or i am doing something wrong here. 
upgraded-methodology-issues yml-talawa

**Comment by palisadoes** [comment]
Please use the example they give. It's changed in the latest version.

- https://github.com/marketplace/actions/auto-label

image

**Comment by VanshikaSabharwal** [comment]
> Please use the example they give. It's changed in the latest version.
> 
> * https://github.com/marketplace/actions/auto-label
> 
> image

Okay i got it. Thank you for explaining .

**Comment by VanshikaSabharwal** [comment]
I have added auto-label.json5 and auto-label.yml files in github/workflows as mentioned in https://github.com/marketplace/actions/auto-label.  Attached is ss of both files. 

(talawa)auto-label yml


image

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please make your changes to the issues.yml file only. The `auto-label.yml` content should be moved to that file and the `auto-label.yml` removed from this PR.

**Comment by VanshikaSabharwal** [comment]
> Please make your changes to the issues.yml file only. The `auto-label.yml` content should be moved to that file and the `auto-label.yml` removed from this PR.

Okay I am doing it. Thank you for telling.

File: .github/workflows/auto-label.json5 [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  labelsSynonyms: {
+    dependencies: ["dependencies", "dependabot", "dependency"],
+    security: ["security", "dependencies", "dependabot", "dependency"],
+    "good first issue": [
+      "security",
+      "dependencies",
```
Comment (palisadoes) [inline .github/workflows/auto-label.json5:7]: 1. These labels don't match what we have
2. You have not followed the guidance of the issue
3. Please fix

**Review by palisadoes — CHANGES_REQUESTED** [review]
You have not followed the guidance of the issue

File: .github/workflows/auto-label.json5 (hunk 1 above)
Comment (VanshikaSabharwal) [reply .github/workflows/auto-label.json5:7]: Apologies for the misunderstanding. I've made the necessary adjustments based on your feedback. The labels now align correctly with the guidance provided. Could you please confirm if everything looks good on your end?

image

**Comment by VanshikaSabharwal** [comment]
Apologies for the misunderstanding. I've made the necessary adjustments based on your feedback. The labels now align correctly with the guidance provided. Could you please confirm @palisadoes , if everything looks good on your end?

image

**Comment by noman2002** [comment]
@VanshikaSabharwal are you still working on this one ?

**Comment by VanshikaSabharwal** [comment]
> @VanshikaSabharwal are you still working on this one ?

Yes @noman2002 i am working on this issue

**Comment by VanshikaSabharwal** [comment]
@palisadoes . @noman2002  I have updated the code according to the guidelines mentioned in the issue. Are there any more changes to be made in the code??

**Comment by palisadoes** [comment]
This PR has too many submitted files. 

Please ensure the following:

1. Fetch the latest upstream/develop files
2. Make your PR modifications based on the local copy of this `develop` branch
3. Submit your PRs against our `develop` branch.
4. Your source branch in your local repository must not be the same as the target branch

Closing

**Pull request closed without merging**